src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/PromptManager.kt

```diff
@@ -245,9 +248,15 @@ class PromptManager(
      * @return The matching PsiMethod if found, otherwise an empty string.
      */
     private fun getPsiMethod(
-        psiClass: PsiClassWrapper,
+        psiClass: PsiClassWrapper?,
         methodDescriptor: String,
     ): PsiMethodWrapper? {
+        // Processing function outside the class
+        if (psiClass == null) {
+            val currentPsiMethod = psiHelper.getSurroundingMethod(caret)!!
+            if (psiHelper.generateMethodDescriptor(currentPsiMethod) == methodDescriptor) return currentPsiMethod
+            return null
```

Comment:
What does this if-statement check for (what logical scenario does it account for)?

Reply:
no need to check it, you are right